src/main/webapp/main.js

```diff
@@ -11,16 +11,16 @@ Vue.component("photo-copyright", {
 Vue.component("search-bar", {
   template: `
   <div id="search-bar">
-    <div id="bar">
-      <input v-on:keyup.enter="submit" v-model="request" type="text" id="text" name="text" placeholder="Search..." />
-      <div v-on:click="submit" id="search-icon"></div>
-    </div>
+    <form id="bar" action="/meal">
```

Comment:
Please undo the `form`, the `action` and the `type=submit`.

This action is done via Javascript already by the means of `fetch` function. Adding it here adds a chance some implementation goes wrong and the request to `/meal` gets submitted twice: once here, and in `fetch` in javascript later.